sdk/src/testIntegration/java/com/hedera/hashgraph/sdk/test/integration/TokenRejectFlowIntegrationTest.java

```diff
@@ -0,0 +1,244 @@
+        var nftTokenIdNftInfo = new TokenNftInfoQuery()
+            .setNftId(nftTokenId.nft(nftSerials.get(1)))
+            .execute(testEnv.client);
+
+        assertThat(nftTokenIdNftInfo.get(0).accountId).isEqualTo(testEnv.operatorId);
+
+        // verify the allowance - should be 0, because TokenRejectFlow dissociates
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> {
+            new TransferTransaction()
+                .addNftTransfer(nftTokenId.nft(nftSerials.get(1)), testEnv.operatorId, receiverAccountId)
+                .execute(testEnv.client)
+                .getReceipt(testEnv.client);
+        }).withMessageContaining("TOKEN_NOT_ASSOCIATED_TO_ACCOUNT");
+
+        new TokenDeleteTransaction()
+            .setTokenId(nftTokenId)
+            .execute(testEnv.client)
+            .getReceipt(testEnv.client);
+
+        testEnv.close(receiverAccountId, receiverAccountKey);
+    }
+
+    @Test
+    @DisplayName("Cannot execute TokenReject flow for NFT when rejecting Only Part Of Owned NFTs")
+    void canExecuteTokenRejectFlowForNftWhenRejectingOnlyPartOfOwnedNFTs() throws Exception {
+        var testEnv = new IntegrationTestEnv(1).useThrowawayAccount();
+        var nftTokenId1 = EntityHelper.createNft(testEnv);
+        var receiverAccountKey = PrivateKey.generateED25519();
+        var receiverAccountId = EntityHelper.createAccount(testEnv, receiverAccountKey, 0);
+
+        var mintReceiptToken = new TokenMintTransaction()
+            .setTokenId(nftTokenId1)
+            .setMetadata(NftMetadataGenerator.generate((byte) 10))
+            .execute(testEnv.client)
+            .getReceipt(testEnv.client);
+
+        var nftSerials = mintReceiptToken.serials;
+
+        // manually associate bft
+        new TokenAssociateTransaction()
+            .setAccountId(receiverAccountId)
+            .setTokenIds(Collections.singletonList(nftTokenId1))
+            .freezeWith(testEnv.client)
+            .sign(receiverAccountKey)
+            .execute(testEnv.client)
+            .getReceipt(testEnv.client);
+
+        // transfer nfts to the receiver
+        new TransferTransaction()
+            .addNftTransfer(nftTokenId1.nft(nftSerials.get(0)), testEnv.operatorId, receiverAccountId)
+            .addNftTransfer(nftTokenId1.nft(nftSerials.get(1)), testEnv.operatorId, receiverAccountId)
+            .execute(testEnv.client)
+            .getReceipt(testEnv.client);
+
+        // execute the token reject flow
+        assertThatExceptionOfType(Exception.class).isThrownBy(() -> {
+            new TokenRejectFlow()
+                .setOwnerId(receiverAccountId)
+                .addNftId(nftTokenId1.nft(nftSerials.get(1)))
+                .freezeWith(testEnv.client)
+                .sign(receiverAccountKey)
+                .execute(testEnv.client)
+                .getReceipt(testEnv.client);
+        }).withMessageContaining("ACCOUNT_STILL_OWNS_NFTS");
```

Comment:
Its expected i think

Reply:
@0xivanov yes. Do you mean that I should update a comment somehow?

Reply:
No, just saying